lib/BackgroundJob/Tasks/ImageProcessingTask.php

```diff
@@ -316,32 +316,24 @@ private function populateDescriptors(\FaceLandmarkDetection $fld, \FaceRecogniti
 		$faces = $imageProcessingContext->getFaces();
 
 		foreach($faces as &$face) {
-			$tempfilePath = $this->cropFace($imageProcessingContext->getImagePath(), $face);
-
-			// Usually, second argument to detect should be just $face. However, since we are doing image acrobatics
-			// and already have cropped image, bounding box for landmark detection is now complete (cropped) image!
-			$landmarks = $fld->detect($tempfilePath, array(
-				"left" => 0, "top" => 0, "bottom" => $face->height(), "right" => $face->width()));
-			$descriptor = $fr->computeDescriptor($tempfilePath, $landmarks);
+			// For each face, we want to detect landmarks and compute descriptors.
+			// We use already resized image (from temp, used to detect faces) for this.
+			// (better would be to work with original image, but that will require
+			// another orientation fix and another save to the temp)
+			// But, since our face coordinates are already changed to align to original image,
+			// we need to fix them up to align them to temp image here.
+			$normalizedFace = clone $face;
```

Comment:
You can use the original without cloning, since it is not used again..
But in any case, can not we normalize the faces to the original image directly when we detect them and work with the normalized ones?

Reply:
Here we do it from the beginning, but should not the ratio be the reverse? Now :confused: 

https://github.com/matiasdelellis/facerecognition/blob/496da698288c7bb2666f36278177c7e01fe8fa36/lib/BackgroundJob/Tasks/ImageProcessingTask.php#L228

Reply:
> You can use the original without cloning

Can I? I was not thinking I could (although my PHP is not good). I am changing fields of `$face` object in-place (in `normalizeSize`) and we use those fields later when we are inserting face in DB (at the end of flow). So, by calling normalize on existing object, values would be changing and we would insert values from resized image, not from original image. Of course, I am talking this out of my ass now:), but here is sample:
https://ideone.com/KPWFs7

> Here we do it from the beginning

This is happening when faces are detected, and before landmark detection. FindFaces is nicely encapsulated to return list of faces with *proper* coordinates (as in original image), so it is hiding (by design) details of resized image on which we are doing face detection. I think it should stay that way. And in method I changed now, I am reversing that ratio that we changed in FindFaces to get us back to resized image coordinates. Not sure if this is what you are asking?